### PR TITLE
Document setting up security rules for the bastion

### LINKED ISSUE #18: No way to set inbound CIDR block for VPC bastion
The `README` seems to indicate that setting `external_access_cidr_block` will allow you to restrict access to the bastion:

```
- external_access_cidr_block - CIDR block for inbound clients to VPC bastion (default: 0.0.0.0/0)
```

However, the security group resource for the bastion doesn't use this variable, and in fact appears to not allow any configuration other than permitting traffic ~from all clients on all ports~ (EDIT: The default behavior when no rules exist is in fact to [disallow all traffic](https://github.com/azavea/terraform-aws-vpc/search?q=external_access_cidr_block&unscoped_q=external_access_cidr_block)):

https://github.com/azavea/terraform-aws-vpc/blob/77d34df313c610688dd295f88c74251baf363f90/main.tf#L129-L137

In fact, the `external_access_cidr_block` variable isn't referenced anywhere except `README.md` and `variables.tf`, as you can see from [this search](https://github.com/azavea/terraform-aws-vpc/search?q=external_access_cidr_block&unscoped_q=external_access_cidr_block).

I would expect the security group resource to include an `ingress` block something like this:

```hcl
ingress {
  from        = 22
  to          = 22
  protocol    = "tcp"
  cidr_blocks = ["${var.external_access_cidr_block}"]
}
```

### PR DESCRIPTION
**Pull request by jeancochrane**
# Overview

The `README` indicates that the value of the variable `external_access_cidr_block` will determine which clients can access the bastion, but the configuration doesn't actually set any inbound rules for the bastion, so no clients will be able to access the bastion no matter how the variable is set (#18). 

~This PR adds an ingress rule to the bastion security group resource, allowing inbound SSH traffic from the CIDR block identified by `external_access_cidr_block`.~

This PR updates the `README` to remove references to the deprecated `external_access_cidr_block` variable, and adds a section on using the module outputs to configure security groups.

## Testing instructions

If you'd like to confirm that the recommended code in the documentation actually works, you can use the [`terraform-config` branch of the `ahoy` breakable toy project](https://github.com/azavea/ahoy/tree/feature/jcochrane/terraform-config).

1. Create a local file `./deployment/terraform/terraform.tfvars` with the following dummy values (replace the IP address and SSH public key with values that you can use to confirm SSH access works later on):

```
postgres_db = "bar"
postgres_host = "baz"
postgres_user = "foo"
postgres_password = ""
django_secret_key = "foobarbaz"
django_allowed_hosts = "localhost"
aws_account_id = "foo"
bastion_inbound_cidr_block = "<your-ip-address-here>"
aws_key_pair_public_key = "<your-public-key-here>"
```

2. If not yet provisioned, provision the VM: `./scripts/setup`
3. SSH into the VM: `vagrant ssh`
5. Run the Terraform container: `docker-compose -f docker-compose.ci.yml run --rm terraform`
6. Set the AWS profile you'd like to use: `export AWS_PROFILE=<your-profile>`
7. Initialize Terraform: `cd ./deployment/terraform && terraform init`
8. Generate a deployment plan: `terraform plan -out terraform.tfplan`
9. Run the plan: `terraform apply terraform.tfplan`
10. Use the output hostname to test inbound SSH: `ssh -i /path/to/key ec2-user@<bastion-dns>`
11. If you'd like, log into the relevant AWS console to confirm that the security groups look good

## Related issues

Closes #18.

### CODE REVIEW

**Comment by hectcastro** [comment]
The `README` is definitely misleading here. I think that the solution here can be limited to `README` tweaks vs. Terraform configuration changes though.

---

Trying to piece the history around this together, we removed all security group rule definitions from the module in https://github.com/azavea/terraform-aws-vpc/commit/02649c65a34549b929ef2b9bc0f851f9af042858 (this is where the `README` updates around `external_access_cidr_block` should have happened).

From there, the idea was that people could use the `bastion_security_group_id` module output to assign any security group rules via `aws_security_group_rule` from _outside_ of the module. We switched to that because we got into trouble making assumptions about what people would and wouldn't want in their security group rules.

Lastly, in https://github.com/azavea/terraform-aws-vpc/commit/1e63a1ff7b3caf7afc891f3683b073599745ac30 we layered in a `aws_network_interface_sg_attachment` resource because we needed a way to associate multiple security groups with the bastion (you can only have so many security group rules associated with a security group, so when you go over that, you need to add more security groups to raise the overall limit).

I think that we can get things into a good state by:

- Removing all `README` and `variables.tf` references to `external_access_cidr_block`
- Adding something like https://github.com/azavea/terraform-aws-ecs-cluster#usage to this module's `README` so that users have a fighting chance

---

> I don't actually know how we test Terraform modules. Any thoughts?

We usually end up testing them within the context of a Terraform project (so you might end up testing this within the context of `ahoy`). Within the Terraform project, we update the [`source`](https://github.com/azavea/ahoy/blob/4e53e66b92a95c57b918068b7f77827bda3aac7d/deployment/terraform/vpc.tf#L2) attribute of the module to be a local path reference or a direct GitHub HTTPS branch reference with a `?ref=feature/jean/enable-inbound-cidr-block` query string.

See: 

- https://www.terraform.io/docs/modules/sources.html
- https://www.terraform.io/docs/modules/sources.html#selecting-a-revision

File: README.md [hunk 1]
```diff
@@ -4,15 +4,27 @@ A Terraform module to create an Amazon Web Services (AWS) Virtual Private Cloud
 
 ## Usage
 
-```javascript
+This module creates a VPC alongside a variety of related resources, including:
+
+- Two public and private subnets
```
[the remark this answers is not in the document]
Comment (jeancochrane) [reply README.md]: Makes sense! There's only ever one Internet Gateway and VPC Endpoint though, correct?

File: main.tf [hunk 2]
```diff
@@ -134,6 +134,13 @@ resource "aws_security_group" "bastion" {
     Project     = "${var.project}"
     Environment = "${var.environment}"
   }
+
+  ingress {
```
[the remark this answers is not in the document]
Comment (jeancochrane) [reply main.tf]: Nice catch! I thought I'd removed this block before pushing the branch. Should be gone now. I didn't actually read those warning boxes on the Terraform docs, that's very helpful context.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hectcastro) [reply README.md]: Yep.

File: main.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (hectcastro) [reply main.tf]: Interesting. I thought I had deleted this comment after I saw that you removed it.

**Pull request merged**